src/canvas-extractor/edfi_canvas_extractor/extract_facade.py

```diff
@@ -192,22 +193,30 @@ def run(arguments: MainArguments) -> None:
     if not succeeded:
         _break_execution("Sections")
 
-    succeeded = _get_assignments(arguments, sync_db)
-    if not succeeded:
-        _break_execution("Assignments")
-
     succeeded = _get_students(arguments, sync_db)
     if not succeeded:
         _break_execution("Students")
 
-    _get_system_activities(arguments, sync_db)
-    _get_submissions(arguments, sync_db)
-
     succeeded = _get_enrollments(arguments, sync_db)
     if not succeeded:
         _break_execution("Enrollments")
 
-    _get_grades(arguments, )  # Grades don't need sync process because they are part of enrollments
-    _get_section_activities(arguments, )  # SectionActivities are empty
+    if Features.Assignments in arguments.feature:
```

Comment:
What do you think about making this a property on the argument object, so that we can get

```
if arguments.ExtractAssignments:
```

This is not very different from a readability standpoint. The main point is lower the class coupling by moving the reference to the constant into the arguments class, thus simplifying the facade just a little bit.

Reply:
✔️